### PR TITLE
Adding support for Qwen model

### PR DESCRIPTION
**Pull request by ariG23498**
A working [colab notebook](https://gist.github.com/ariG23498/9a7aea9da39378116a74d803be84f89b)

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
-def infer(vllm_engine, inputs):
+def infer(model_name, vllm_engine, inputs):
     sampling_params = inputs.pop("sampling_params")
-    vllm_inputs = [{"prompt": PROMPT, "multi_modal_data": {"image": image}} for image in inputs["original_images"]]
+    vllm_inputs = [
+        {
+            "prompt": PROMPT_MAP[model_name],
+            "multi_modal_data": {"image": image}
+        } for image in inputs["original_images"]
+    ]
```
Comment (sayakpaul) [inline model.py]: I prefer a line-length of 119.

File: model.py [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+    "llava": LLAVA_CKPT_ID,
+    "qwen-vl": QWEN_VL_CKPT_ID,
+}
+PROMPT_MAP = {
+    "llava": LLAVA_PROMPT,
+    "qwen-vl": QWEN_VL_PROMPT,
+}
 
-def load_vllm_engine(max_tokens=120):
+def load_vllm_engine(model_name, max_tokens=120):
```
Comment (sayakpaul) [inline model.py]: We should default to a model_name (`llava` in this case), otherwise, this is breaking change.

File: main.py [hunk 3]
```diff
@@ -11,14 +11,18 @@
 
 
 def main(
+    model_name: str,
```
Comment (sayakpaul) [inline main.py]: `model_name` should have a default, otherwise, it's a breaking change.

File: config.py [hunk 4]
```diff
@@ -27,9 +30,45 @@
 a. Present the final caption as a single, well-formed sentence or paragraph.
 b. Ensure the caption is clear, concise, and accurately represents the image content.
 
-Remember, your captions will be used to train a text-to-image generation model, so strive for accuracy and completeness in your descriptions.
+Remember, your captions will be used to train a text-to-image generation model, so strive for accuracy and completeness in your descriptions. [/INST]"""
```
Comment (sayakpaul) [inline config.py:33]: Nice catch.

File: utils.py [hunk 5]
```diff
@@ -21,6 +21,7 @@ def save_results(output_queue, output_dir):
 
             if len(item) == 4:
                 original_captions, outputs, img_bytes, img_hashes = item
+                watermark_scores = None
```
Comment (sayakpaul) [inline utils.py:24]: Good catch!

**Comment by sayakpaul** [comment]
Can you clear the outputs of the Colab and show the outputs files and some of its contents?

**Review by sayakpaul — COMMENTED** [review]
Left some comments. Thanks for contributing!

Can you clear the outputs of the Colab and show the outputs files and some of its contents?

**Comment by ariG23498** [comment]
> Can you clear the outputs of the Colab and show the outputs files and some of its contents?

I am trying to create a smaller webdataset to iterate faster and showcase results. In case you already have something like that, would you like to share?

**Comment by sayakpaul** [comment]
> I am trying to create a smaller webdataset to iterate faster and showcase results. In case you already have something like that, would you like to share?

Unfortunately, I don't have it :(

**Comment by ariG23498** [comment]
## Command to run

```shell
$ python3 main.py\
--data_path="https://huggingface.co/datasets/pixparse/cc3m-wds/resolve/main/cc3m-train-0000.tar"\
--model_name="qwen-vl"
```

## Output

example

```
{
    "original": "the family has been a part .",
    "predicted": "The image shows a family of four, consisting of a man, a woman, and two children, sitting on the front steps of their house. The family is dressed in matching white tops, and they appear to be posing for a photo. The house has a white door and siding, and there is a black mat on the steps."
}
```

File: config.py [hunk 6]
```diff
@@ -27,9 +30,17 @@
...
+####################################################################################################
+# QWEN VL
+####################################################################################################
+QWEN_VL_PROMPT = """\
+<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n\
+<|im_start|>user\n<|vision_start|><|image_pad|><|vision_end|>\
+f"What is in the image?<|im_end|>\n\
+<|im_start|>assistant\n"""
+
+QWEN_VL_CKPT_ID = "Qwen/Qwen2-VL-7B-Instruct"
```
Comment (sayakpaul) [inline config.py]: Add a note that this prompt might not be optimized for captions for text-to-image generation.

**Review by sayakpaul — APPROVED** [review]
LGTM! Thanks! 

Let's add a note that this repo currently supports Qwen2-VL too!

**Review by sayakpaul — APPROVED** [review]

**Comment by sayakpaul** [comment]
Thanks for your contributions!

**Pull request merged**